core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -1170,14 +1172,14 @@ private[history] case class LogInfo(
     appId: Option[String],
     attemptId: Option[String],
     fileSize: Long,
-    lastIndex: Option[Long],
+    @JsonDeserialize(contentAs = classOf[JLong]) lastIndex: Option[Long],
```

Comment:
This is basically the same approach used elsewhere right? if so that's fine.

Reply:
Yes right. There're other spots we already took care of.